### PR TITLE
ENH add osx virt package patch

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This PR adds the osx virtual package patches.

### CODE REVIEW

**Comment by beckermr** [comment]
@isuruf for viz and review

relates to https://github.com/conda-forge/conda-forge.github.io/issues/891

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -368,6 +369,20 @@ def _fix_libgfortran(fn, record):
...
+def _add_osx_virt(fn, record):
+    rconst = record.get("run_constrained", ())
+    dep_idx = next(
+        (q for q, dep in enumerate(rconst)
+         if dep.split(' ')[0] == "__osx"),
+        None
+    )
+    run_constrained = list(rconst)
+    if dep_idx is None:
+        run_constrained.append("__osx >=10.9")
```
Comment (isuruf) [inline recipe/gen_patch_json.py]: We don't need to add the 10.9 as it's implied by the miniconda installer. We should add this constraint for the few feedstocks that use one that's greater than 10.9

Comment (beckermr) [reply recipe/gen_patch_json.py]: ahhhh crap. I will redo the PR.

**Comment by beckermr** [comment]
OK fixed @isuruf

**Comment by isuruf** [comment]
Here's the actual list.
```
qt/conda-forge/osx-64/qt-5.12.1-h1b46049_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
qt/conda-forge/osx-64/qt-5.9.7-h8cf7e54_3.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
qt/conda-forge/osx-64/qt-5.9.7-h93ee506_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
qt/conda-forge/osx-64/qt-5.9.7-h93ee506_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
qt/conda-forge/osx-64/qt-5.12.5-h1b46049_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
qt/conda-forge/osx-64/qt-5.9.7-h93ee506_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/osx-64/openmpi-mpicxx-4.0.1-h6052eea_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/osx-64/openmpi-mpicxx-4.0.1-h6052eea_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/osx-64/openmpi-mpicxx-4.0.1-h6052eea_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/linux-64/openmpi-mpicxx-4.0.1-hc9558a2_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/linux-64/openmpi-mpicxx-4.0.1-hc9558a2_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicxx/conda-forge/linux-64/openmpi-mpicxx-4.0.1-hc9558a2_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
nodejs/conda-forge/osx-64/nodejs-12.8.0-hec2bf70_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.1.0-h6de7cb9_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.3.1-h6de7cb9_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.9.0-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.9.1-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.7.0-hec2bf70_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.10.0-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.4.0-h6de7cb9_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.11.1-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.7.0-h6de7cb9_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.3.0-h6de7cb9_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-10.16.3-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.12.0-hfddbe92_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
nodejs/conda-forge/osx-64/nodejs-12.8.1-hec2bf70_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.10",
freecad/conda-forge/osx-64/freecad-0.18.3-py37h4764a83_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
freecad/conda-forge/osx-64/freecad-0.18.3-py37hc453731_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
freecad/conda-forge/osx-64/freecad-0.18.4-py37hab2b3aa_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
freecad/conda-forge/osx-64/freecad-0.18.4-py37hab2b3aa_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/osx-64/openmpi-mpicc-4.0.1-h24e1f75_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/osx-64/openmpi-mpicc-4.0.1-h24e1f75_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/osx-64/openmpi-mpicc-4.0.1-h24e1f75_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/linux-64/openmpi-mpicc-4.0.1-h516909a_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/linux-64/openmpi-mpicc-4.0.1-h516909a_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpicc/conda-forge/linux-64/openmpi-mpicc-4.0.1-h516909a_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
javafx-sdk/conda-forge/osx-64/javafx-sdk-11.0.4-h6dcaf97_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.11",
javafx-sdk/conda-forge/osx-64/javafx-sdk-12.0.2-h6dcaf97_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.11",
javafx-sdk/conda-forge/osx-64/javafx-sdk-12.0.2-h6dcaf97_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.11",
javafx-sdk/conda-forge/osx-64/javafx-sdk-11.0.4-h6dcaf97_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.11",
openmpi-mpifort/conda-forge/osx-64/openmpi-mpifort-4.0.1-h939af09_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/osx-64/openmpi-mpifort-4.0.1-h6ad152f_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/osx-64/openmpi-mpifort-4.0.1-h939af09_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/osx-64/openmpi-mpifort-4.0.1-h939af09_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/linux-64/openmpi-mpifort-4.0.1-he991be0_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/linux-64/openmpi-mpifort-4.0.1-he991be0_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi-mpifort/conda-forge/linux-64/openmpi-mpifort-4.0.1-he991be0_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
reaktoro/conda-forge/osx-64/reaktoro-1.0.7-py37h99eb986_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
reaktoro/conda-forge/osx-64/reaktoro-1.0.7-py37h99eb986_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
reaktoro/conda-forge/osx-64/reaktoro-1.0.7-py36h99eb986_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
reaktoro/conda-forge/osx-64/reaktoro-1.0.7-py36h99eb986_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py38he22c54c_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py37h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py36he22c54c_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py27h2a560b1_4.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py27h2a560b1_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py37h2a560b1_4.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py36h2a560b1_3.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py27h2a560b1_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py36h2a560b1_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py27h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py37h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py27he22c54c_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py27he22c54c_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py37h2a560b1_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py37h2a560b1_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py36h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py36h2a560b1_4.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py27h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py37h2a560b1_3.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py38he22c54c_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py27h2a560b1_3.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.9.2-py36h2a560b1_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py37he22c54c_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py36he22c54c_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py37he22c54c_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
pyqt/conda-forge/osx-64/pyqt-5.12.3-py36h2a560b1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
ldas-tools-al/conda-forge/osx-64/ldas-tools-al-2.6.3-hf543496_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
ldas-tools-al/conda-forge/osx-64/ldas-tools-al-2.6.3-hf543496_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
ldas-tools-al/conda-forge/osx-64/ldas-tools-al-2.6.4-h4f290e7_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
ldas-tools-al/conda-forge/osx-64/ldas-tools-al-2.6.4-h4f290e7_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/osx-64/openmpi-4.0.1-ha90c164_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/osx-64/openmpi-4.0.1-ha90c164_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/osx-64/openmpi-4.0.1-hfcebdee_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/osx-64/openmpi-4.0.1-ha90c164_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/linux-64/openmpi-4.0.1-hc99cbb1_1.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/linux-64/openmpi-4.0.1-hc99cbb1_0.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
openmpi/conda-forge/linux-64/openmpi-4.0.1-hc99cbb1_2.json:  "MACOSX_DEPLOYMENT_TARGET": "10.12",
```

**Comment by beckermr** [comment]
Is your proposal to simply patch those entries directly? I think besides openmpi, the code I have will work.

**Comment by isuruf** [comment]
Some previous versions of qt can be installed in 10.9, so we don't want to impose 10.12 on them.

**Comment by beckermr** [comment]
OK. I checked the repo index data manually and this appears correct. FYI, the `run_constrains` key in meta.yaml turns into just `constrains` in the index, which was a fun discovery. I also had to adjust the diff code to add new keys. It still does not support deleting keys via diffing. We can add this if we want.

**Comment by mbargull** [comment]
> FYI, the `run_constrains` key in meta.yaml turns into just `constrains` in the index, which was a fun discovery.

Yep, `build`/`host`/`run`-everything is only on the build side. For `conda` only `run` dependencies matter, hence no need to clarify that :D.

**Pull request merged**